File: leetcode/easy/easy.14/easy.14.ts
export function longestCommonPrefix(strs: string[]): string {
  if (strs.length === 0) return "";

  const sortedStrs = strs.sort((a, b) => a.localeCompare(b));
  const [first, last] = [sortedStrs[0], sortedStrs[sortedStrs.length - 1]];

  let prefix = "";

  for (let i = 0; i < first.length; i++) {
    if (first[i] !== last[i]) {
      return prefix;
    }

    prefix += first[i];
  }

  return prefix;
}

File: leetcode/easy/easy.14/easy.14.go
package easy14

import "sort"

func longestCommonPrefix(strs []string) string {
	sort.Strings(strs)
	
	first := strs[0]
	last := strs[len(strs) - 1]

	prefix := ""

	for i := 0; i < len(first); i++ {
		if first[i] != last[i] {
			return prefix
		}

		prefix += string(first[i])
	}

	return prefix
}